pkg/daemon/update.go

```diff
@@ -589,32 +583,33 @@ func getFileOwnership(file ignv2_2types.File) (int, int, error) {
 
 // Update a given PasswdUser's SSHKey
 func (dn *Daemon) updateSSHKeys(newUsers []ignv2_2types.PasswdUser) error {
-	// Keys should only be written to "/home/core/.ssh"
-	// Once Users are supported fully this should be writing to PasswdUser.HomeDir
-	if newUsers[0].Name != "core" {
-		// Double checking that we are only writing SSH Keys for user "core"
-		return fmt.Errorf("Expecting user core. Got %s instead", newUsers[0].Name)
-	}
-	sshDirPath := filepath.Join("/home", newUsers[0].Name, ".ssh")
-	// we are only dealing with the "core" User at this time, so only dealing with the first entry in Users[]
-	glog.Infof("Writing SSHKeys at %q", sshDirPath)
-	if err := dn.fileSystemClient.MkdirAll(filepath.Dir(sshDirPath), os.FileMode(0600)); err != nil {
-		return fmt.Errorf("Failed to create directory %q: %v", filepath.Dir(sshDirPath), err)
-	}
-	glog.V(2).Infof("Created directory: %s", sshDirPath)
-
-	authkeypath := filepath.Join(sshDirPath, "authorized_keys")
-	var concatSSHKeys string
-	for _, k := range newUsers[len(newUsers)-1].SSHAuthorizedKeys {
-		concatSSHKeys = concatSSHKeys + string(k) + "\n"
-	}
+	if len(newUsers) >= 1 {
```

Comment:
LGTM, though (optional) I would tend to do e.g.:

```
if len(newUsers) == 0 {
    return nil
}
```

To avoid re-indenting, and also on the general principle of getting the special cases out of the way in a function first, leaving the main body of the function to handle the general case.

Reply:
will do!

Reply:
Done. Thanks for this comment!